examples/reporting/get_campaign_stats_to_csv.py

```diff
@@ -0,0 +1,152 @@
+          segments.date DURING LAST_7_DAYS
+        ORDER BY metrics.impressions DESC
+        LIMIT 25
+        """
+
+    # Issues a search request using streaming.
+    response = ga_service.search_stream(customer_id, query)
+    try:
+        with open(output_file, "w", newline="") as f:
+            writer = csv.writer(f)
+
+            # Define a list of headers for the first row.
+            headers = ["Account", "Date", "Campaign", "Impressions", "Clicks",
+                       "Cost"]
+
+            # If the write_headers flag was passed, write header row to the CSV
+            if write_headers:
+                writer.writerow(headers)
+
+            for batch in response:
+                for row in batch.results:
+                    # Use the CSV writer to write the individual GoogleAdsRow
+                    # fields returned in the SearchGoogleAdsStreamResponse.
+                    writer.writerow(
+                        [
+                            row.customer.descriptive_name,
+                            row.segments.date,
+                            row.campaign.name,
+                            row.metrics.impressions,
+                            row.metrics.clicks,
+                            row.metrics.cost_micros,
+                        ]
+                    )
+
+            print(f"Customer {customer_id} report written to {output_file}")
+
+    except GoogleAdsException as ex:
+        print(
+            f'Request with ID "{ex.request_id}" failed with status '
+            f'"{ex.error.code().name}" and includes the following errors:'
+        )
+        for error in ex.failure.errors:
+            print(f'\tError with message "{error.message}".')
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print(f"\t\tOn field: {field_path_element.field_name}")
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    # GoogleAdsClient will read the google-ads.yaml configuration file in the
+    # home directory if none is specified.
+    google_ads_client = GoogleAdsClient.load_from_storage()
+
+    parser = argparse.ArgumentParser(
+        description="Retrieves a campaign stats and writes to CSV file."
+    )
+    # The following argument(s) should be provided to run the example.
+    parser.add_argument(
+        "-c",
+        "--customer_id",
+        type=str,
+        required=True,
+        help="The Google Ads customer ID of the account you would like to get "
+             "the report for to write to CSV.",
+    )
+    parser.add_argument(
+        "-o",
+        "--output_file",
+        type=str,
+        required=True,
+        help="Name of the local CSV file to save the report to. File will be "
+             "saved in the same directory as the script.",
+    )
+    # Optional boolean argument for writing headers.
+    parser.add_argument(
+        "-w",
+        "--write_headers",
+        action="store_true",
```

Comment:
The comment in the main method indicates that this parameter is False if not provided, but `store_true` implies the default value is `True`, should this be `"store_false"`?

Reply:
This is the behavior as documented by the author: https://stackoverflow.com/a/8203679/2801259

> The store_true option automatically creates a default value of False.
> Likewise, store_false will default to True when the command-line argument is not present.

I think changing to store_false and checking in the code for the absence of True would be even more confusing? I've updated the comment for now, but happy to do it another way as per your advice.  
```
            if not write_headers:
                writer.writerow(headers)
```

Reply:
Ah, in that case I think the current implementation makes more sense - I was confused `store_true` makes me think the value `True` is stored and used by default,  but it's the other way around.